controllers/networkchaos/netem/types.go

```diff
@@ -254,15 +266,47 @@ func (r *Reconciler) applyPod(ctx context.Context, pod *v1.Pod, networkchaos *v1
 
 	containerID := pod.Status.ContainerStatuses[0].ContainerID
 
-	netem, err := spec.ToNetem()
-	if err != nil {
-		return err
-	}
-
 	_, err = pbClient.SetNetem(ctx, &pb.NetemRequest{
 		ContainerId: containerID,
 		Netem:       netem,
 	})
 
 	return err
 }
+
+// mergeNetem calls ToNetem on all non nil network emulation specs and merges them into one request.
+func mergeNetem(spec v1alpha1.NetworkChaosSpec) (*pb.Netem, error) {
+	// NOTE: a cleaner way like
+	// emSpecs = []NetemSpec{spec.Delay, spec.Loss} won't work.
+	// Because in the for _, spec := range emSpecs loop,
+	// spec != nil would always be true.
+	// See https://stackoverflow.com/questions/13476349/check-for-nil-and-nil-interface-in-go
+	// And https://groups.google.com/forum/#!topic/golang-nuts/wnH302gBa4I/discussion
+	// > In short: If you never store (*T)(nil) in an interface, then you can reliably use comparison against nil
+	var emSpecs []NetemSpec
+	if spec.Delay != nil {
+		emSpecs = append(emSpecs, spec.Delay)
+	}
+	if spec.Loss != nil {
+		emSpecs = append(emSpecs, spec.Loss)
+	}
+	if spec.Duplicate != nil {
+		emSpecs = append(emSpecs, spec.Duplicate)
+	}
+	if spec.Corrupt != nil {
+		emSpecs = append(emSpecs, spec.Corrupt)
+	}
+	if len(emSpecs) == 0 {
+		return nil, errors.New(invalidNetemSpecMsg)
+	}
+
+	merged := &pb.Netem{}
+	for _, spec := range emSpecs {
+		em, err := spec.ToNetem()
+		if err != nil {
+			return nil, err
+		}
+		utils.MergeNetem(merged, em)
```

Comment:
```
merged = utils.MergeNetem(merged, em)
```

I think the source of your bug is here. @at15 

In the chaos-daemon log, it doesn't receive correct merged netem.

Reply:
Thank!, I get it now I am not using my own function properly, it should be

`merged = utils.MergeNetem(merged, em)` ....

Reply:
Should get fixed now

```
2020-03-23T06:17:10.002Z	INFO	chaos-daemon-server	Set netem	{"Request": "netem:<time:90000 jitter:90000 delay_corr:25 loss:25 loss_corr:25 > container_id:\"containerd://08311decd2944bf9ef43f2031acb475b66c7ef11ee4a726786884fad1ddd27b8\" "}

64 bytes from 10.244.1.4: icmp_seq=23 ttl=62 time=0.136 ms

46 packets transmitted, 40 received, 13% packet loss, time 45489ms
```